java-frontend/src/test/files/se/NonNullSetToNullCheck.java

```diff
@@ -0,0 +1,101 @@
+package javax.annotation;
+
+@interface Nonnull {}
+
+public class MainClass {
+  
+  @Nonnull
+  private String primary;
+  private String secondary;
+  
+  public MainClass(String color) {
+    if (color != null) {
+      secondary = null;
+    }
+    primary = color;  // Noncompliant {{"primary" is marked "javax.annotation.Nonnull" but is set to null.}}
+  }
+  
+  public MainClass(@Nonnull String color, String other) {
+    primary = color;
+    secondary = other;
+  }
+  
+  public MainClass() { // Noncompliant  {{"primary" is marked "javax.annotation.Nonnull" but is not initialized in this constructor.}}
+    return; // Just for coverage
+  }
+  
+  public @Nonnull String colorMix() {
```

Comment:
Usually we place the annotation related to the return type of the method before the modifiers. See for instance: https://github.com/SonarSource/sonar-java/blob/master/java-frontend/src/main/java/org/sonar/java/se/ProgramState.java#L327